lib/rubocop/ast/node.rb

```diff
@@ -624,6 +627,15 @@ def parent_module_name_for_block(ancestor)
         end
       end
 
+      def and_return?(node)
+        return unless node.and_type?
+
+        rhs = node.rhs
+
+        %i[return break next].include?(rhs.type) ||
+          rhs.send_type? && rhs.method?(:raise) || rhs.method?(:fail)
```

Comment:
I think there might be an operator precedence problem here, but I'm not sure.

I assume you mean:

```ruby
rhs.send_type? && (rhs.method?(:raise) || rhs.method?(:fail))
```

but Ruby does this:

```ruby
(rhs.send_type? && (rhs.method?(:raise)) || rhs.method?(:fail))
```

Reply:
Would a node pattern like this work?

```
(and _ {return break next (send _ {:raise :fail})})
```

Reply:
I was too slow! 😄

Reply:
Ah, I pushed the changes with a time lag 😅